oshmem/mca/atomic/ucx/atomic_ucx_module.c

```diff
@@ -32,6 +32,95 @@ int mca_atomic_ucx_finalize(void)
     return OSHMEM_SUCCESS;
 }
 
+static inline
+int mca_atomic_ucx_op(void *target,
+                      const void *value,
+                      size_t size,
+                      int pe,
+                      ucp_atomic_post_op_t op)
+{
+    ucs_status_t status;
+    ucs_status_ptr_t status_ptr;
+    spml_ucx_mkey_t *ucx_mkey;
+    uint64_t rva;
+    uint64_t val;
+
+    if (8 == size) {
+        val = *(uint64_t*)value;
+    } else if (4 == size) {
+        val = *(uint32_t*)value;
+    } else {
+        ATOMIC_ERROR("[#%d] Type size must be 4 or 8 bytes.", my_pe);
+        return OSHMEM_ERROR;
+    }
+
+    ucx_mkey = mca_spml_ucx_get_mkey(pe, target, (void *)&rva);
+    status = ucp_atomic_post(mca_spml_self->ucp_peers[pe].ucp_conn,
+                             op, val, size, rva,
+                             ucx_mkey->rkey);
+    return ucx_status_to_oshmem(status);
+}
+
+static inline
+int mca_atomic_ucx_fop(void *target,
+                       void *prev,
+                       const void *value,
+                       size_t size,
+                       int pe,
+                       ucp_atomic_fetch_op_t op)
+{
+    ucs_status_t status;
+    ucs_status_ptr_t status_ptr;
+    spml_ucx_mkey_t *ucx_mkey;
+    uint64_t rva;
+    uint64_t val;
+
+    if (8 == size) {
+        val = *(uint64_t*)value;
+    } else if (4 == size) {
+        val = *(uint32_t*)value;
+    } else {
+        ATOMIC_ERROR("[#%d] Type size must be 4 or 8 bytes.", my_pe);
+        return OSHMEM_ERROR;
+    }
+
+    ucx_mkey = mca_spml_ucx_get_mkey(pe, target, (void *)&rva);
+    status_ptr = ucp_atomic_fetch_nb(mca_spml_self->ucp_peers[pe].ucp_conn,
+                                     op, val, prev, size,
+                                     rva, ucx_mkey->rkey,
+                                     opal_common_ucx_empty_complete_cb);
+    status = opal_common_ucx_wait_request(status_ptr, mca_spml_self->ucp_worker);
```

Comment:
looks like status is checked twice here: in opal_common_ucx_wait_request and in opal_common_ucx_wait_request
can we eliminate the extra branch? maybe opal_common_ucx_wait_request() should return opal_status_t?